src/scripts/download-new-songs.ts

```diff
@@ -0,0 +1,96 @@
+import * as ytdl from "ytdl-core";
+import * as fs from "fs";
+import * as _config from "../../config/app_config.json";
+import * as mysql from "promise-mysql";
+import { QueriedSong } from "types";
+import * as path from "path";
+let config: any = _config;
+
+export function clearPartiallyCachedSongs() {
+    console.log("Clearing partially cached songs");
+    if (!fs.existsSync(config.songCacheDir)) {
+        return console.error("Song cache directory doesn't exist.");
+    }
+    fs.readdir(config.songCacheDir, (error, files) => {
+        if (error) {
+            return console.error(error);
+        }
+
+        const endingWithPartRegex = new RegExp("\\.part$");
+        const partFiles = files.filter((file) => file.match(endingWithPartRegex));
+        partFiles.forEach((partFile) => {
+            fs.unlink(`${config.songCacheDir}/${partFile}`, (err) => {
+                if (err) {
+                    console.error(err);
+                }
+            })
+        })
+        if (partFiles.length) {
+            console.log(`${partFiles.length} stale cached songs deleted.`);
+        }
+    });
+}
+
+const downloadSong = (id: string) => {
+    let cachedSongLocation = path.join(config.songCacheDir, `${id}.mp3`);
+    const tempLocation = `${cachedSongLocation}.part`;
+    let cacheStream = fs.createWriteStream(tempLocation);
+    const ytdlOptions = {
+        filter: "audioonly" as const,
+        quality: "highest"
```

Comment:
Can this be as const as well?

Reply:
The `const` cast is only required because `ytdlOptions.filter` requires type `Filter` which is defined as `'audioandvideo' | 'video' | 'videoonly' | 'audio' | 'audioonly' | ((format: videoFormat) => boolean);`, it will not compile otherwise

Reply:
Understood, thanks!